src/qz/installer/TaskControl.java

```diff
@@ -0,0 +1,180 @@
+            String[] procs = ShellUtilities.executeRaw("jcmd", "-l").split("\\r?\\n");
+            for(String proc : procs) {
+                String[] parts = proc.split(" ", 1);
+                if (parts.length >= 2 && parts[1].contains(PrintSocketServer.class.getCanonicalName())) {
+                    killCmd[killCmd.length - 1] = parts[0].trim();
+                    success = success && ShellUtilities.execute(killCmd);
+                }
+            }
+        }
+
+        if(SystemUtilities.isWindowsXP()) {
+            File f = new File("TempWmicBatchFile.bat");
+            if(f.exists()) {
+                f.deleteOnExit();
+            }
+        }
+
+        return success;
+    }
+
+    public static ArrayList<Path> locateProcessPaths(boolean exactMatch, String ... ProcessNames) throws IOException {
+        ArrayList<Path> pathList = new ArrayList<>();
+        ArrayList<String> pidArray = getRootPIDs(exactMatch, ProcessNames);
+        if (SystemUtilities.isWindows()) {
+            for(String pid : pidArray) {
+                WIN32_PATH_QUERY[WIN32_PATH_QUERY_INPUT_INDEX] = "ProcessId=" + pid;
+                String[] response = ShellUtilities.executeRaw(WIN32_PATH_QUERY).split("\\s*\\r?\\n");
+                if (response.length > 1) {
+                    pathList.add(Paths.get(response[1]).toRealPath());
+                }
+            }
+        } else {
+            for(String pid : pidArray) {
+                pathList.add(Paths.get("/proc/", pid, "/exe").toRealPath());
+            }
+        }
+        return pathList;
+    }
+
+    public static ArrayList<String> getRootPIDs(boolean exactMatch, String ... ProcessNames) {
+        String[] response;
+        ArrayList<String> pidList = new ArrayList<>();
+
+        if (ProcessNames.length == 0) return pidList;
+
+        if (SystemUtilities.isWindows()) {
+            ArrayList<String> parentPIDList = new ArrayList<>();
+            String matchPartial = exactMatch ? "Name='%s'" : "Name like '/%%s/%'";
+
+            String matchString = "(";
+            for (int i = 0; i < ProcessNames.length; i++) {
+                if (i != 0 ) matchString += " OR ";
+                matchString += String.format(matchPartial, ProcessNames[i]);
+            }
+            matchString += ")";
+
+            WIN32_PID_QUERY[WIN32_PID_QUERY_INPUT_INDEX] = matchString;
+            response = ShellUtilities.executeRaw(WIN32_PID_QUERY).split("[\\r\\n]+");
+
+            // Skip the first result (the first row is column headers)
+            for (int i = 1; i < response.length; i++) {
+                String[] row =  response[i].split("[\\s,]+");
+
+                parentPIDList.add(row[0]);
+                pidList.add(row[1]);
+            }
+
+            // Remove all PIDs that have a parent of the same name
+            for (int i = pidList.size() - 1; i >= 0; i--){
+                if (pidList.contains(parentPIDList.get(i))) {
+                    pidList.remove(i);
```

Comment:
Why not [`removeAll()`](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#removeAll-java.util.Collection-), the header shouldn't cause issues.

Reply:
Answering my own question, it's an odd parent/orphan relationship.  Code stays, per @Vzor-'s explanation via phone call :D